client/src/components/PortfolioTable.tsx

```diff
@@ -0,0 +1,741 @@
+              value={table.getState().pagination.pageSize}
+              onChange={(e) => {
+                table.setPageSize(Number(e.target.value));
+              }}
+            >
+              {[10, 20, 50, 100, 500].map((pageSize) => (
+                <option key={pageSize} value={pageSize}>
+                  Show {pageSize}
+                </option>
+              ))}
+            </select>
+          </div>
+          <div className="next">
+            <button
+              className="page-btn"
+              onClick={() => table.nextPage()}
+              disabled={!table.getCanNextPage()}
+            >
+              {i18n._(t`Next`)}
+            </button>
+          </div>
+        </div>
+      </div>
+    );
+  })
+);
+
+const renderContacts = ({ row }: { row: Row<AddressRecord> }) => {
+  return (
+    <ul className="contacts-list">
+      {Object.entries(_groupBy(row.original.allcontacts, "value"))
+        .sort(sortContactsByImportance)
+        .map((group) => group[1][0])
+        .map((contact, i) => (
+          <li key={i}>
+            {contact.title}: {contact.value}
+          </li>
+        ))}
+    </ul>
+  );
+};
+
+type ArrowIconProps = {
+  dir: "up" | "down" | "both";
+};
+
+function ArrowIcon(props: ArrowIconProps) {
+  return (
+    <span className="arrow-icon">
+      {props.dir === "up" ? (
+        <span>↑</span>
+      ) : props.dir === "down" ? (
+        <span>↓</span>
+      ) : (
+        <>
+          ↑<span>↓</span>
+        </>
+      )}
+    </span>
+  );
+}
+
+function formatCurrency(x: number | null): string | null {
+  return x == null ? null : currencyFormater.format(x);
+}
+
+export function isPartOfGroupSale(saleId: string, addrs: AddressRecord[]) {
+  const addrsWithMatchingSale = addrs.filter((addr) => addr.lastsaleacrisid === saleId);
+  return addrsWithMatchingSale.length > 1;
+}
+
+function formatAbatementStartYear(year: number | null, i18n: I18n) {
+  const thisYear = new Date().getFullYear();
+  if (!year) return null;
+  return year > thisYear ? i18n._(t`Starts ${year}`) : i18n._(t`Since ${year}`);
+}
+
+function findMostCommonType(complaints: HpdComplaintCount[] | null) {
+  return complaints && complaints.length > 0 && complaints[0].type;
+}
+
+// TODO: This is getting triggered by sorts on these columns
+function useFilterOptionsUpdater(
+  filterContext: IFilterContext,
+  setFilterContext: React.Dispatch<React.SetStateAction<IFilterContext>>,
+  table: Table<AddressRecord>
+) {
+  const ownernamesOptionValues = table.getColumn("ownernames").getFacetedUniqueValues();
+  const unitsresOptionValues =
+    table.getColumn("unitsres").getFacetedMinMaxValues() || MINMAX_DEFAULT;
+  const zipOptionValues = table.getColumn("zip").getFacetedUniqueValues();
+
+  React.useEffect(() => {
+    setFilterContext({
+      ...filterContext,
+      filterOptions: {
+        ownernames: Array.from(new Set(Array.from(ownernamesOptionValues.keys()).flat())).sort(
+          compareAlphaNumLast
```

Comment:
just for my own clarity, what are the cases with numeric names?

Reply:
oh yeah, and that's probably worth adding a comment. Most of the corporations start with numbers (`123 fake st, LLC`), and after talking to Joel we thought it made sense to just put them last so there are some real person names in the list at first so it's less confusing.